ZCS-1661 Add support for publishing zip as artifact from any repo

silentsakky opened the pull request:
zm-zcs:ant-global.xml
   - Create a new target zimbra-zip which can be used to publish zips instead of jars from ivy
   - By default jar target will be calling zimbra-jar but caller can override and call zimbra-zip which will publish a zip file
   - removed deploy-foss and publish-foss-local targets as those are related to zm-mailbox which is not using this file currently

zm-zcs:ivysettings.xml
   - when using ivy:install target sometime it was not resolving to proper zip files, so as per suggestion in post http://grokbase.com/t/ant/ivy-user/086ct4mr6h/newbie-cant-resolve-zip-files, I have added ivy pattern line in local resolver

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Zimbra/zm-zcs?pullRequest=5) <br/>All committers have signed the CLA.

fatkudu commented on ant-global.xml:
```diff
@@ -586,20 +619,4 @@
     <target name="dist.dependencies">
          <antcall target="depend.${ant.project.name}"/>
     </target>
```
Please use a different commit for this unrelated change. (The deletion of publish-foss-local and deploy-foss targets)

silentsakky replied:
Thanks
created separate commit for this change, not adding any jira ticket number as this change is not reported anywhere

silentsakky committed:
ZCS-1661 Add support for publishing zip as artifact from any repo

ant-global.xml
   - Create a new target zimbra-zip which can be used to publish zips instead of jars from ivy
   - By default jar target will be calling zimbra-jar but caller can override and call zimbra-zip which will publish a zip file

ivysettings.xml
   - when using ivy:install target sometime it was not resolving to proper zip files, so as per suggestion in post http://grokbase.com/t/ant/ivy-user/086ct4mr6h/newbie-cant-resolve-zip-files, I have added ivy pattern line in local resolver
   - create a separate resolver for zip files, as changing default resolver was giving errors

silentsakky committed:
removed deploy-foss and publish-foss-local targets as those are related to zm-mailbox which is not using this file anymore
